dspace-api/src/main/java/org/dspace/app/requestitem/RequestItemServiceImpl.java

```diff
@@ -117,15 +122,18 @@ public void delete(Context context, RequestItem requestItem) {
     }
 
     @Override
-    public boolean isRestricted(Context context, DSpaceObject o)
-            throws SQLException {
-        List<ResourcePolicy> policies = authorizeService
-                .getPoliciesActionFilter(context, o, Constants.READ);
+    public boolean isRestricted(Context context, DSpaceObject o, EPerson person) throws SQLException {
+        List<ResourcePolicy> policies = authorizeService.getPoliciesActionFilter(context, o, Constants.READ);
+
         for (ResourcePolicy rp : policies) {
-            if (resourcePolicyService.isDateValid(rp)) {
+            boolean isSameEPerson = rp.getEPerson() != null && rp.getEPerson().equals(person);
+            boolean isGroupMember = rp.getGroup() != null && groupService.isMember(context, person, rp.getGroup());
+
+            if (resourcePolicyService.isDateValid(rp) && (isSameEPerson || isGroupMember)) {
                 return false;
             }
         }
+
         return true;
```

Comment:
it seems to me that this whole method is unnecessary from the start (not a fault of this PR).
Should we just use the https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/authorize/service/AuthorizeService.java#L68
in the `org.dspace.app.requestitem.RequestItemEmailNotifier.sendResponse(Context, RequestItem, String, String)` removing this method from the `RequestItemService` interface?  @tdonohue what do you think?

Reply:
Replaced RequestItemService.isRestricted() with AuthorizeService.authorizeActionBoolean()